docs/gds-supported-platform/getting-started-gsp.md

```diff
@@ -0,0 +1,167 @@
+
+## Create a Kubernetes Deployment object
+
+You run an app by creating a [Kubernetes Deployment object](https://kubernetes.io/docs/concepts/#kubernetes-objects). This object defines your app and its routes, databases and all other relevant information. You describe a Deployment in a YAML file.
+
+1. Create a `deployment.yaml` file in the `templates` directory. The following example uses an [nginx](https://hub.docker.com/_/nginx/) container image called `myapp`. Replace this nginx container image with your app image:
+
+    ```
+    apiVersion: apps/v1beta2
+    kind: Deployment
+    metadata:
+      name: {{ .Release.Name }}-myapp
+      labels:
+        app.kubernetes.io/name: myapp
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      replicas: {{ .Values.replicas }}
+      selector:
+        matchLabels:
+          app.kubernetes.io/name: myapp
+          app.kubernetes.io/instance: {{ .Release.Name }}
+      template:
+        metadata:
+          labels:
+            app.kubernetes.io/name: myapp
+            app.kubernetes.io/instance: {{ .Release.Name }}
+        spec:
+          containers:
+            - name: myappcontainer
+              image: "nginx:latest" #Replace this with your app image
+              ports:
+                - name: http
+                  containerPort: 80
+                  protocol: TCP
+    ```
+
+    Helm automatically populates the `{{ .Release.Name }}` and `{{ .Values.replicas }}` variables when you render the chart.
+
+1. Run the following command in the root directory to render the chart:
+
+    ```
+    helm template --name example .
+    ```
+
+1. Check `stdout` to see if the chart rendered correctly.
+
+## Create a service
+
+By default, your apps are not accessible to the public. To expose them to the public, you must set up a [Service](https://kubernetes.io/docs/concepts/services-networking/service/) and an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) into the Kubernetes cluster.
+
+Setting up a Service creates a stable endpoint to send public internet traffic to. To set up a service, create a `service.yaml` file in the `templates` directory with the following code:
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Release.Name }}-APP_NAME
+  labels:
+    app.kubernetes.io/name: APP_NAME
+    app.kubernetes.io/instance: {{ .Release.Name }}
+spec:
+  type: ClusterIP
+  ports:
+    - port: 80
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app.kubernetes.io/name: APP_NAME
+    app.kubernetes.io/instance: {{ .Release.Name }}
+```
+Helm automatically populates the `{{ .Release.Name }}` variable when you render the chart.
+
+## Create an Ingress
+
+You must define an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) to route public internet traffic to the stable endpoint you created when you set up the Service.
```

Comment:
Also not quite true. A `Service` of type `LoadBalancer` would also be routable from the public Internet. An Ingress is not required. I guess it depends on the target audience and framing of this documentation. This is more a "if you want to hook into some of the GSP's magic, and your service is HTTPS, then you can simplify the public traffic routing by creating an `Ingress`". 🤷‍♂

Reply:
agreed, but I think this one is _ok_ for now ... it is a very light skimming of the concepts and the aim of this task is to touch on the three abstractions required to expose a micro-service to the public.